### PR TITLE
moving plot's sub and add methods to abstract class

### LINKED ISSUE #236: missing argument in __sub__
we're not implementing `__sub__` in the plots. [example](https://github.com/ploomber/sklearn-evaluation/blob/20ae8ebfc449df54c18ce87fce988317d57d2d16/src/sklearn_evaluation/plot/roc.py#L234), but we're raising an exception so users explicitly know it won't work.

however, the signature is incorrect; we're missing one argument. otherwise, we get this error (instead of the NotimplementedError one):

```
__sub__() takes 1 positional argument but 2 were given
TypeError: __sub__() takes 1 positional argument but 2 were given
```

to simplify things, we should remove `__sub__` implementations from concrete classes to the `AbstractPlot` class so all plots inherit from it

### PR DESCRIPTION
**Pull request by WSShawn**
## Describe your changes
- Added a generic implementation for plot __sub__ and __add__ method. 
- Added unit tests for plot __sub__ and __add__ method.

## Issue ticket number and link
Closes #236 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://sklearn-evaluation--320.org.readthedocs.build/en/320/

### CODE REVIEW

File: tests/test_roc.py [hunk 1]
```diff
@@ -270,18 +270,6 @@ def test_roc_dump_multi(tmp_directory, roc_multi_classification_raw_data):
     assert roc._get_data() == roc2._get_data()
 
 
-@pytest.mark.parametrize(
```
[the remark this answers is not in the document]
Comment (WSShawn) [reply tests/test_roc.py:273]: yes, I put it in a separate test file, is this not what we want?

**Comment by WSShawn** [comment]
CI passed! Sorry for all the back and forth on this issue!

**Comment by WSShawn** [comment]
@edublancas request for review!

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
great!

**Pull request merged**